utils/http.go

```diff
@@ -112,6 +176,21 @@ func buildAccessRecords(repo string, actions ...string) []auth.Access {
 	return requiredAccess
 }
 
+// buildCatalogRecord returns the only valid format for the catalog
+// resource. Only admins can get this access level from the token
+// server.
+func buildCatalogRecord(actions ...string) []auth.Access {
+	requiredAccess := []auth.Access{{
+		Resource: auth.Resource{
+			Type: "registry",
```

Comment:
This is basically the only valid form for the catalog record as far as Docker's token server is concerned. Docker's token server will only ever return a catalog record with these Resource field values.